src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java

```diff
@@ -385,91 +385,92 @@ public void writeTo(DownloadInstance di, Class<?> clazz, Type type, Annotation[]
 
                 } 
 
-                InputStream instream = storageIO.getInputStream();
-                if (instream != null) {
-                    // headers:
-
-                    String fileName = storageIO.getFileName();
-                    String mimeType = storageIO.getMimeType();
-
-                    // Provide both the "Content-disposition" and "Content-Type" headers,
-                    // to satisfy the widest selection of browsers out there. 
-                    // Encode the filename as UTF-8, then deal with spaces. "encode" changes
-                    // a space to + so we change it back to a space (%20).
-                    String finalFileName = URLEncoder.encode(fileName, "UTF-8").replaceAll("\\+", "%20");
-                    httpHeaders.add("Content-disposition", "attachment; filename=\"" + finalFileName + "\"");
-                    httpHeaders.add("Content-Type", mimeType + "; name=\"" + finalFileName + "\"");
-
-                    long contentSize;
-                    boolean useChunkedTransfer = false;
-                    //if ((contentSize = getFileSize(di, storageIO.getVarHeader())) > 0) {
-                    if ((contentSize = getContentSize(storageIO)) > 0) {
-                        logger.fine("Content size (retrieved from the AccessObject): " + contentSize);
-                        httpHeaders.add("Content-Length", contentSize);
-                    } else {
-                        //httpHeaders.add("Transfer-encoding", "chunked");
-                        //useChunkedTransfer = true;
-                    }
+                try (InputStream instream = storageIO.getInputStream()) {
```

Comment:
Looks good to me. try-with-resources is the preferred way to assure streams are closed these days and that's what the PR does.